perception/euclidean_cluster/lib/voxel_grid_based_euclidean_cluster.cpp

```diff
@@ -117,7 +117,7 @@ bool VoxelGridBasedEuclideanCluster::cluster(
       voxel_grid_.getCentroidIndexAt(voxel_grid_.getGridCoordinates(point.x, point.y, point.z));
     if (map.find(index) != map.end()) {
       auto & cluster_data_size = clusters_data_size.at(map[index]);
-      if (cluster_data_size + point_step > std::size_t(max_cluster_size_ * point_step)) {
+      if (cluster_data_size > std::size_t(max_cluster_size_ * point_step)) {
```

Comment:
@badai-nguyen  Do you have any intention to not writing like this?

```suggestion
      if (cluster_data_size >= std::size_t(max_cluster_size_ * point_step)) {
```

Reply:
With this change, we also need to fix line139 to not filter out clusters with max_cluster_size

```
static_cast<int>(i_cluster_data_size / point_step) <= max_cluster_size_)) {
```

Reply:
@YoshiRi  if change as [your suggestion](https://github.com/autowarefoundation/autoware.universe/pull/7734#discussion_r1662105558)
we need to change `static_cast<int>(i_cluster_data_size / point_step) <= max_cluster_size_)) {`  
to `static_cast<int>(i_cluster_data_size / point_step) < max_cluster_size_)) {`. 
In the result, the actual biggest cluster can be output will have `max_cluster_size_ -1` points. 
Otherwise, it will cause the same issue as current (as orrange case: also output a parth of cluster-A which actually is bigger than `max_cluster_size_')
 
![Untitled Diagram](https://github.com/autowarefoundation/autoware.universe/assets/94814556/e170f991-fc01-4e16-bba0-5529c2b0d06f)